assembl/static2/js/app/components/common/avatarImage.jsx

```diff
@@ -1,19 +1,31 @@
+// @flow
 import React from 'react';
 
-const AvatarImage = ({ userId, userName }) => {
-  const src = userId ? `/user/id/${userId}/avatar/30` : '/static2/img/icons/avatar.png';
-  return (
-    <img
-      src={src}
-      width="30"
-      height="30"
-      alt={userName}
-      onError={(e) => {
-        const target = e.target;
-        target.src = '/static2/img/icons/avatar.png';
-      }}
-    />
-  );
+import { getIconPath } from '../../utils/globalFunctions';
+
+type Props = {
+  userId: ?(number | string),
+  userName: ?string
 };
 
+class AvatarImage extends React.PureComponent<Props> {
```

Comment:
Curious: what is the advantage of a PureComponent over a functional component?

Reply:
https://moduscreate.com/blog/react_component_rendering_performance/

Reply:
functional component is wrapped by a component

Reply:
maintenant nous avons le  React.memo Hook pour optimiser les "functional component"

Reply:
a few warnings about pure components though: https://reactjs.org/docs/react-api.html#reactpurecomponent

Reply:
```React.PureComponent’s shouldComponentUpdate() only shallowly compares the objects. If these contain complex data structures, it may produce false-negatives for deeper differences.```

So this is indeed the best option here since the props are very basic :)